torch/_dynamo/testing.py

```diff
@@ -246,24 +245,33 @@ def __call__(
         return gm.forward
 
 
-# Equivalent to backend="aot_eager", but also records graphs that
-# we can assert on
-class AOTEagerAndRecordGraphs:
+class AotEagerAndRecordGraphs:
     def __init__(self) -> None:
         self.graphs: List[torch.fx.GraphModule] = []
+        self.fw_graphs: List[torch.fx.GraphModule] = []
+        self.bw_graphs: List[torch.fx.GraphModule] = []
 
     def __call__(
         self, gm: torch.fx.GraphModule, example_inputs: List[torch.Tensor]
     ) -> Callable[..., Any]:
-        def save_graph(gm: torch.fx.GraphModule, *args: Any, **kwargs: Any) -> Any:
-            self.graphs.append(gm)
+        from torch._dynamo.backends.common import aot_autograd
+
+        self.graphs.append(gm)
+
+        def fw_compiler(
+            gm: torch.fx.GraphModule, example_inputs: List[torch.Tensor]
+        ) -> Callable[..., Any]:
+            self.fw_graphs.append(gm)
+            return gm.forward
+
+        def bw_compiler(
+            gm: torch.fx.GraphModule, example_inputs: List[torch.Tensor]
+        ) -> Callable[..., Any]:
+            self.bw_graphs.append(gm)
             return gm.forward
 
-        return aot_eager(
-            gm,
-            example_inputs,
-            fw_compiler=save_graph,
-            bw_compiler=save_graph,
+        return aot_autograd(fw_compiler=fw_compiler, bw_compiler=bw_compiler)(
+            gm, example_inputs
```

Comment:
It seems a bit odd that this class is named `AotEagerAndRecordGraphs` yet uses `aot_autograd` rather than `aot_eager` here, is it on purpose?

Reply:
`aot_eager` uses min-cut partitioner partitioner that I was trying to avoid here. But no strong reason.